include/ck_tile/ops/epilogue/cshuffle_epilogue.hpp

```diff
@@ -51,18 +53,19 @@ struct CShuffleEpilogue
     using ODataType   = remove_cvref_t<typename Problem::ODataType>;
     using BTypeToUse =
         std::conditional_t<std::is_same_v<BDataType, pk_int4_t>, ODataType, BDataType>;
-    using CLayout                           = remove_cvref_t<typename Problem::CLayout>;
-    static constexpr index_t kBlockSize     = Problem::kBlockSize;
-    static constexpr index_t kMPerBlock     = Problem::kMPerBlock;
-    static constexpr index_t kNPerBlock     = Problem::kNPerBlock;
-    static constexpr index_t kMWave         = Problem::kMWave;
-    static constexpr index_t kNWave         = Problem::kNWave;
-    static constexpr index_t kMPerXdl       = Problem::kMPerXdl;
-    static constexpr index_t kNPerXdl       = Problem::kNPerXdl;
-    static constexpr index_t kKPerXdl       = Problem::kKPerXdl;
-    static constexpr index_t isCTransposed  = Problem::isCTransposed;
-    static constexpr index_t kMPerIteration = kMPerXdl * kMWave;
-    static constexpr index_t kNPerIteration = kNPerXdl * kNWave;
+    using CLayout = remove_cvref_t<typename Problem::CLayout>;
+    static constexpr memory_operation_enum MemoryOperation = Problem::MemoryOperation;
```

Comment:
If we put this information as a class member, you no longer need to have it as a `operator()` template paramter.

Reply:
I still see this as a tparam here: 
https://github.com/ROCm/composable_kernel/pull/2001/files#diff-a2466bfef61d871813a8a210b406e7db886cc294935562fe0076017c0e7f75aaR127